binary/private/binary_disk.f90

```diff
@@ -0,0 +1,303 @@
+      real(dp), parameter :: the_grid_max = 1.0_dp
+      real(dp) :: the_grid(n_the)  ! grid for disk thickness
+      real(dp) :: T_arr(n_the)
+      real(dp), parameter :: T_floor = 3.0d3  ! [K] the minimum value for disk temperature solution
+      real(dp) :: T, T_max, the, the_left, the_right, the_min, the_max, separation_factor, dlogthe
+      real(dp) :: T_left, T_right, f1, f1_left, f2, f2_left, f2_right, f, f_left
+      real(dp) :: logT_arr(n_the), logthe_grid(n_the)
+
+      ierr = 0
+
+      ! key parameters
+      M2 = accretor_mass * Msun
+      M1dot = mass_transfer_rate * Msun/secyer
+      a = orbital_separation * Rsun
+      q = accretor_mass / donor_mass  ! mass ratio M2/M1
+
+      log_q = log10(q)
+
+      ! positions of Lagrangian points (based on analytic fits)
+      xL1 = -0.0355_dp * log_q**2 + 0.251_dp * abs(log_q) + 0.500_dp  ! [a = SMA]
+      xL2 = 0.0756_dp * log_q**2 - 0.424_dp * abs(log_q) + 1.699_dp  ! [a]
+
+      if (log_q > 0.0_dp) then  ! m2 is more massive
+         xL1 = 1.0_dp - xL1
+         xL2 = 1.0_dp - xL2
+      end if
+      mu = q / (1.0_dp + q)
+
+      ! outer disk radius
+      Rd_over_a = (1.0_dp - xL1)**4 / mu
+      ! relavent potential energies
+      PhiL1_dimless = -((1.0_dp - mu)/abs(xL1) + mu/abs(1.0_dp - xL1) + 0.5*(xL1 - mu)**2)  ! [G(M1+M2)/a]
+      PhiL2_dimless = -((1.0_dp - mu)/abs(xL2) + mu/abs(1.0_dp - xL2) + 0.5*(xL2 - mu)**2)  ! [G(M1+M2)/a]
+      PhiRd_dimless = -(1.0_dp - mu + mu/Rd_over_a + 0.5*(1.0_dp - mu)**2)
+
+      GM2 = standard_cgrav * M2
+
+      Rd = Rd_over_a*a
+      Phi_units = standard_cgrav * (M2 / mu) / a
+      PhiL1 = PhiL1_dimless * Phi_units
+      PhiL2 = PhiL2_dimless * Phi_units
+      PhiRd = PhiRd_dimless * Phi_units
+      ! Keplerian frequency at Rd
+      omega_K = sqrt(GM2 / Rd**3)
+
+
+      ! constants involved in numerical solutions
+      c1 = 2.0_dp * pi * crad * disk_alpha * Rd / (3.0_dp * omega_K * M1dot)
+      c2 = boltzm * Rd / (GM2 * disk_mu * mp)
+      c3 = 8.0_dp * pi**2 * crad * disk_alpha * clight * Rd**2 / (M1dot**2 * omega_K)
+      c4 = 2.0_dp * pi * disk_mu * crad * disk_alpha * omega_K * mp * Rd**3 / (boltzm * M1dot)
+
+      ! Create logarithmically spaced grid for grid search for disk thickness [only used at the beginning]
+      do i = 1, n_the
+         the_grid(i) = 10.0_dp**(log10(the_grid_min) + (i - 1) * (log10(the_grid_max) - log10(the_grid_min)) / (n_the - 1))
+      end do
+
+      ! only T < T_max is possible to calculate
+      T_max = pow(4.0_dp / (27.0_dp * c1**2 * c2), 1.0_dp / 9.0_dp)
+
+      do i = 1, n_the
+         T_arr(i) = 0.0_dp
+      end do
+
+      do i = 1, n_the
+         the = the_grid(i)
+         ! use bisection method
+         T_left = 0.1_dp * min(the**2 / c2, T_max)
+         f1_left = f1_the_T_fL2(the, T_left, 0.0_dp, c1, c2)
+         T_right = T_max
+         do while (abs((T_left - T_right) / T_right) > tol)
+            T = (T_left + T_right) / 2.0_dp
```

Comment:
There are many of these and I'm not sure if there's consensus but I prefer
```suggestion
            T = 0.5_dp * (T_left + T_right)
```
because when not optimising, some compilers (at least `gfortran`) don't optimise the division into a multiplication. This probably doesn't improve performance but it removes ambiguity.